lib/mixlib/shellout/windows/core_ext.rb

```diff
@@ -308,25 +309,66 @@ def create(args)
           end
 
           unless bool
-            raise SystemCallError.new("CreateProcessAsUserW (You must hold the 'Replace a process level token' permission)", FFI.errno)
+            raise SystemCallError.new("CreateProcessAsUserW (You must hold the 'Replace a process level token' permission. Restart your machine after adding this right to make it effective.)", FFI.errno)
           end
         else
-          bool = CreateProcessWithLogonW(
-            logon,                  # User
-            domain,                 # Domain
-            passwd,                 # Password
-            LOGON_WITH_PROFILE,     # Logon flags
-            app,                    # App name
-            cmd,                    # Command line
-            hash["creation_flags"], # Creation flags
-            env,                    # Environment
-            cwd,                    # Working directory
-            startinfo,              # Startup Info
-            procinfo                # Process Info
-          )
+          if hash["elevated"]
+            token = FFI::MemoryPointer.new(:ulong)
+
+            bool = LogonUserW(
+              logon,                      # User
+              domain,                     # Domain
+              passwd,                     # Password
+              LOGON32_LOGON_BATCH,        # Logon Type
+              LOGON32_PROVIDER_DEFAULT,   # Logon Provider
+              token                       # User token handle
+            )
 
-          unless bool
-            raise SystemCallError.new("CreateProcessWithLogonW", FFI.errno)
+            unless bool
+              raise SystemCallError.new("LogonUserW", FFI.errno)
```

Comment:
We should check if the error is `ERROR_LOGON_TYPE_NOT_GRANTED ` and return a typed exception with a message telling the user what right the user needs.